azurerm/resource_arm_virtual_machine_scale_set.go

```diff
@@ -658,6 +658,24 @@ func resourceArmVirtualMachineScaleSet() *schema.Resource {
 
 			"tags": tagsSchema(),
 		},
+
+		CustomizeDiff: func(diff *schema.ResourceDiff, v interface{}) error {
+
+			//os_profile.0.admin_password is required unless os_profile_linux_config.0.disable_password_authentication is true
+			if _, hasPassword := diff.GetOk("os_profile.0.admin_password"); !hasPassword {
+				disablePassword := false
+
+				if v, hasDisable := diff.GetOk("os_profile_linux_config.0.disable_password_authentication"); hasDisable {
+					disablePassword = v.(bool)
+				}
+
+				if !disablePassword {
+					return fmt.Errorf(" `os_profile.0.admin_password` must be set unless `os_profile_linux_config.0.disable_password_authentication` is true")
+				}
+			}
+
+			return nil
+		},
```

Comment:
Admin Password is a field which (IME) is almost always interpolated outside of simple scenarios - in which case this won't work, as such can we remove the CustomizeDiff here?